chef/content_distribution
# set up jobs to create archives and generate presigned urls

Signed-off-by: rishichawda <rishichawda@users.noreply.github.com>



## Description

This PR contains set up for jobs to run every Sunday. One main job creates one job for each directory and will download, generate an archive, upload it to a separate private bucket and get a pre-signed url which is valid for 15mins by default. Max can be one week for the latest spec (according to docs). I have added notes within the comments for some changes that may be required later on.

To test it on local, you'll need to:

1. Add an `.envrc` file in `src/content_distribution` directory with the following content:
```bash
export AWS_ACCESS_KEY_ID="*****************" # access key
export AWS_SECRET_ACCESS_KEY="**********************************" # secret key
export ORIGIN_BUCKET="*****************" # source bucket to pull the content
export UPSTREAM_BUCKET="*****************" # dest. bucket where archives are uploaded
```
2. In the `config/schedule.yml` file, update the `cron` key to `*/1 * * * *` to run every minute.
3. Run `sidekiq` in the terminal to run the jobs. 

Please make sure the redis server is running in the default port (6379) before running sideqik command.

## Related Issue



## Types of changes

- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:


- [ ] I have run the pre-merge tests locally and they pass.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).
